### PR TITLE
[ready-for-core-team-review]CRM-16189, added function to create deferred transactions

### PR DESCRIPTION
**Pull request by pradpnayak**
---
- CRM-16189: Improve support for Accrual Method bookkeeping
  https://issues.civicrm.org/jira/browse/CRM-16189

### CODE REVIEW

**Comment by pradpnayak** [comment]
depends on https://github.com/civicrm/civicrm-core/pull/8577

**Comment by colemanw** [comment]
@pradpnayak or @Edzelopez can you please rebase this one?

**Comment by Edzelopez** [comment]
Hi @colemanw, I've rebased the PR.

**Comment by Edzelopez** [comment]
jenkins, retest this please

**Comment by colemanw** [comment]
Now I'm confused. I'm seeing the same test failure here as #8582 

     3128 - Failure: api_v3_SyntaxConformanceTest::testCreateSingleValueAlter with data set #21 ('FinancialAccount')

**Comment by Edzelopez** [comment]
Yeah, I tried running api_v3_SyntaxConformanceTest on my local and it passed there :/

File: CRM/Core/BAO/FinancialTrxn.php [hunk 1]
```diff
@@ -561,6 +561,7 @@ public static function getPartialPaymentTrxn($contribution, $params) {
   }
 
   /**
+<<<<<<< HEAD
```
Comment (totten) [inline CRM/Core/BAO/FinancialTrxn.php]: # 😜

Comment (Edzelopez) [inline CRM/Core/BAO/FinancialTrxn.php]: Whoops!

**Comment by Edzelopez** [comment]
jenkins, retest this please

**Pull request merged**